lib/rails/dom/testing/assertions/selector_assertions.rb

```diff
@@ -263,6 +263,29 @@ def assert_dom_email(&block)
         end
         alias_method :assert_select_email, :assert_dom_email
 
+
+        # Changes the depth of the document_root_element for the operations
+        # nested within the provided block. Accepts a singular Nokogiri node or
+        # an Nokogiri NodeSet as its argument.
+        #
+        #  within 'div[class~=?]', 'hello', count: 1 do
+        #    assert_dom 'p', 'Hello!'
+        #  end
+        #
+        #  within document_root_element.at_css("main") do
+        #    assert_dom "h1", "Hello, World"
+        #  end
+        def within(selector_or_selection, *arguments, **options, &block)
+          case selector_or_selection
+          when nil
+            raise ArgumentError, "Expecting a Node, NodeSet, or selector as the first argument" if selector_or_selection.nil?
```

Comment:
```suggestion
            raise ArgumentError, "Expecting a Node, NodeSet, or selector as the first argument"
```

Don't need the tailing-if that's handled by the `case` statement.